### PR TITLE
Merge upstream intel realsense ros2 branch

### PR DESCRIPTION
**Pull request by pepisg**
Not tested yet. Will post an update in the comments when I test this

**UPDATE**: Now tested. See comments below

### CODE REVIEW

File: realsense2_camera/CMakeLists.txt [hunk 1]
```diff
@@ -130,12 +130,16 @@ elseif("$ENV{ROS_DISTRO}" STREQUAL "eloquent")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DELOQUENT")
   set(SOURCES "${SOURCES}" src/ros_param_backend_foxy.cpp)
 elseif("$ENV{ROS_DISTRO}" STREQUAL "foxy")
-  message(STATUS "Build for ROS2 foxy")
+  message(STATUS "Build for ROS2 Foxy")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DFOXY")
   set(SOURCES "${SOURCES}" src/ros_param_backend_foxy.cpp)
 elseif("$ENV{ROS_DISTRO}" STREQUAL "galactic")
-  message(STATUS "Build for ROS2 galactic")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DFOXY")
```
Comment (pepisg) [inline realsense2_camera/CMakeLists.txt:138]: @charlielito not sure about why this flag was set to `DFOXY` Here. Was this on purpose?

**Comment by pepisg** [comment]
@charlielito I tested this on local and on a robot and confirmed that the following is working:
**LOCAL**:

- [x] RGB Image (checked and local console and rviz)
- [x] Pointcloud (checked in rviz)
- [x] depth image (checked in rviz)
- [x] Get coords service (checke in local console)
- [x] Get pixel service (checked in local console with c2c)
- [x] Get version service (checked in cli)

**ROBOT**
(I built against librealsense v2.48.0, we were using 2.45.0)
- [x] RGB Image (checked in console)
- [x] Pointcloud (checked in cli)
- [x] depth image (checked in cli)
- [x] Get coords service (checked in console with waypoint, visual measure and c2c)
- [x] Get pixel service (checked inl console with c2c)
- [x] Get version service (checked in cli)
- [x] Imu data (Pitch calibrated successfully and imu data was generated properly)

File: realsense2_camera/CMakeLists.txt (hunk 1 above)
Comment (charlielito) [reply realsense2_camera/CMakeLists.txt:138]: No, it is a mistake :sweat_smile:

**Review by charlielito — APPROVED** [review]
Looks nice!

**Pull request merged**